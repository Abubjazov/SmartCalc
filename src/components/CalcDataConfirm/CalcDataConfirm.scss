@import '../../variables';

.calc-data-confirm {
	display: grid;
	grid-template-columns: 300px;
	grid-template-rows: 50px 40px auto auto;
	grid-row: auto;
	row-gap: 10px;
	justify-items: center;
	animation-name: opacity;
	animation-duration: 700ms;

	ul {
		margin-top: 13px;
		justify-self: center;
		font-size: 20px;
		font-weight: 300;

		li {
			width: 200px;
			text-align: center;
			border-bottom: 1px solid $main-color;
		}
	}

	button {
		padding: 0 13px;
		border: none;
		border-radius: 20px;
		min-width: 100px;
		height: 40px;
		text-align: center;
		font-size: 14px;
		line-height: 14px;
		background-color: $main-color;
		color: $white;
		transition: all 0.1s linear;
		outline: none;
		cursor: pointer;

		&:hover,
		&:focus {
			background-color: $main-color-hover;
		}
	}

	&-search {
		margin-top: 13px;
		display: flex;
		justify-content: space-between;

		.sort {
			width: 24px;
			height: 24px;
			padding-left: 4.5px;
			padding-top: 0px;
			background-color: $main-color;
			color: $white;
			font-size: 26px;
			line-height: 31px;
			border-radius: 50px;

			&:hover {
				background-color: $main-color-hover;
			}
		}

		.rotated {
			transform: rotate(180deg);
		}
	}

	&-footer {
		width: 90%;
		margin-top: 31px;
		display: flex;
		justify-content: space-between;
	}
}
