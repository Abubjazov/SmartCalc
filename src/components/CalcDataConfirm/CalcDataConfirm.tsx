import { nanoid } from 'nanoid'
import { ChangeEvent, useState } from 'react'
import { NavLink } from 'react-router-dom'
// import { useActions } from '../../hooks/useActions'
// import { useTypedSelector } from '../../hooks/useTypedSelector'
import { matchCheck } from '../../utils/utils'

import './CalcDataConfirm.scss'

export const CalcDataConfirm = (): JSX.Element => {
	const [items] = useState<number[]>([12, 11, 13])
	const [sortedItems, setSortedItems] = useState<number[]>([12, 11, 13])
	const [search, setSearch] = useState<string>('')
	const [sortDirection, setSortDirection] = useState<boolean>(true)

	const sortItems = () => {
		if (sortDirection) {
			setSortedItems(sortedItems.sort((a, b) => a - b))
			setSortDirection(false)
		} else {
			setSortedItems(sortedItems.sort((a, b) => b - a))
			setSortDirection(true)
		}
	}

	const changeHandler = (e: ChangeEvent<HTMLInputElement>) => {
		setSearch(e.target.value)

		setSortedItems(items.filter(item => matchCheck(item, e.target.value)))
	}

	return (
		<div className='calc-data-confirm'>
			<h1>Подтвердите данные</h1>
			<div className='calc-data-confirm-search'>
				<input
					type='text'
					placeholder='Поиск'
					onChange={changeHandler}
					value={search}
				/>

				<div
					className={`sort ${sortDirection ? null : 'rotated'}`}
					onClick={sortItems}
				>
					^
				</div>
			</div>

			<ul>
				{sortedItems.map(item => (
					<li key={nanoid()}>{item}</li>
				))}
			</ul>

			<div className='calc-data-confirm-footer'>
				<NavLink to='/smartcalc/calc'>Назад</NavLink>
				<NavLink to='/smartcalc/calc'>Подтверждаю</NavLink>
			</div>
		</div>
	)
}

// const [items] = useState<number[]>([12, 11, 13])
// 	const [sortedItems, setSortedItems] = useState<number[]>([12, 11, 13])
// 	const [search, setSearch] = useState<string>('')
// 	const [sortDirection, setSortDirection] = useState<boolean>(true)

// 	const sortItems = () => {
// 		if (sortDirection) {
// 			setSortedItems(sortedItems.sort((a, b) => a - b))
// 			setSortDirection(false)
// 		} else {
// 			setSortedItems(sortedItems.sort((a, b) => b - a))
// 			setSortDirection(true)
// 		}
// 	}

// 	const changeHandler = (e: ChangeEvent<HTMLInputElement>) => {
// 		setSearch(e.target.value)

// 		setSortedItems(items.filter(item => matchCheck(item, e.target.value)))
// 	}

// 	return (
// 		<div className='calc-data-confirm'>
// 			<h1>Подтвердите данные</h1>
// 			<div className='calc-data-confirm-search'>
// 				<input
// 					type='text'
// 					placeholder='Поиск'
// 					onChange={changeHandler}
// 					value={search}
// 				/>

// 				<div
// 					className={`sort ${sortDirection ? null : 'rotated'}`}
// 					onClick={sortItems}
// 				>
// 					^
// 				</div>
// 			</div>

// 			<ul>
// 				{sortedItems.map(item => (
// 					<li key={nanoid()}>{item}</li>
// 				))}
// 			</ul>

// 			<div className='calc-data-confirm-footer'>
// 				<NavLink to='/smartcalc/calc'>Назад</NavLink>
// 				<NavLink to='/smartcalc/calc'>Подтверждаю</NavLink>
// 			</div>
// 		</div>
// 	)
