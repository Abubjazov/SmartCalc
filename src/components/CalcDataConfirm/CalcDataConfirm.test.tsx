import { render } from '@testing-library/react'
import { CalcDataConfirm } from './CalcDataConfirm'

describe('Component: Calculator', () => {
	test('should render Calculator', () => {
		const { asFragment } = render(<CalcDataConfirm />)

		expect(asFragment()).toMatchSnapshot()
	})
})
