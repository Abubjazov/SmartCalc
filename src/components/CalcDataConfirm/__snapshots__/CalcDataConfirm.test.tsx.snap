// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CalcDataConfirm should render CalcDataConfirm by default 1`] = `
<DocumentFragment>
  <div
    class="calc-data-confirm"
  >
    <h1>
      Подтвердите данные
    </h1>
    <div
      class="calc-data-confirm-search"
    >
      <input
        placeholder="Поиск"
        type="text"
        value=""
      />
      <div
        class="sort rotated"
        tabindex="0"
      >
        ^
      </div>
    </div>
    <ul>
      <li>
        2
      </li>
      <li>
        3
      </li>
      <li>
        1
      </li>
      <li>
        5
      </li>
    </ul>
    <div
      class="calc-data-confirm-footer"
    >
      <button>
        Назад
      </button>
      <button>
        Подтверждаю
      </button>
    </div>
  </div>
</DocumentFragment>
`;
